index.js

```diff
@@ -1,97 +1,135 @@
+        responseResult.push(this.books);
         let newDiv = document.createElement('div');
-        newDiv.innerHTML = `<div class="panel panel-success"><div class="panel-heading"><h1 class="text-center">${data[key].summary.title}` +
-            `</h1></div><div class="panel-body"><div class="col-md-2"><img src="${data[key].img}` +
-            `"width="100%"  class="img-responsive"></img></div><div class="col-md-8"><h4 class="text-justify">${data[key].description}` +
-            `</h4></div><div class="col-md-2"><button class="btn btn-info openModal" id=${key}` +
+        newDiv.innerHTML = `<div class="panel panel-success"><div class="panel-heading"><h1 class="text-center">${this.books.summary.title}` +
+            `</h1></div><div class="panel-body"><div class="col-md-2"><img src="${this.books.img}` +
+            `"width="100%"  class="img-responsive"></img></div><div class="col-md-8"><h4 class="text-justify">${this.books.description}` +
+            `</h4></div><div class="col-md-2"><button class="btn btn-info openModal" data-id=${this.books.id}` +
             `>Show<br>summary</button></div></div></div>`;
         document.querySelector('#bookshelf').appendChild(newDiv);
-    }
+    };
 };
-let books = {};
-let serch;
-let startIndex = 0;
+
 function getBooks(src) {
-    $.ajax({
-        url: src,
-        dataType: 'json',
-        success: (data) => {
-            for (let i = 0; i < data.items.length; i++) {
-                let {
-                    etag: id,
-                    volumeInfo: {
-                        title,
-                        categories,
-                        publisher,
-                        publishedDate,
-                        authors,
-                        description,
-                        imageLinks: {
-                            thumbnail: img,
-                        }
-                    }
-                } = data.items[i];
-                books[id] = {
-                    summary: {
-                        title: title || 'Unknown',
-                        author: authors || 'Unknown',
-                        category: categories || 'Unknown',
-                        publisher: publisher || 'Unknown',
-                        published_date: publishedDate || 'Unknown'
-                    },
-                    description: description || 'Unknown',
-                    img: img || 'Unknown'
-                };
-            };
-            createBlock(books);
-        },
-        type: 'GET'
+    return new Promise((resolve, reject) => {
+        let xhr = new XMLHttpRequest();
+        xhr.open('GET', src);
+        xhr.responseType = 'json';
+        xhr.addEventListener('load', () => {
+            resolve(xhr.response);
+        });
+        xhr.send();
     });
 };
+
+document.getElementById('btnSerch').addEventListener('click', (e) => {
+    serch = document.getElementById('bookName').value;
+    if (document.querySelector('div.panel-success')) {
```

Comment:
You can use `if (document.querySelector('div.panel-success')) clean()` in case one-string-short operation